iothub/service/src/AmqpServiceClient.cs

```diff
@@ -37,21 +37,25 @@ internal sealed class AmqpServiceClient : ServiceClient
 
         private int _sendingDeliveryTag;
 
-        public AmqpServiceClient(IotHubConnectionString iotHubConnectionString, bool useWebSocketOnly, ServiceClientTransportSettings transportSettings, ServiceClientOptions options)
+        public AmqpServiceClient(
```

Comment:
Does this constructor need to stay public? This class appears to be internal

Reply:
Public types on an internal class are effectively internal. It is not uncommon to see this.

Reply:
yes. Usually the access modifiers are changed on the class. That way when you want to change visibility it is easy. That is the pattern I've mostly seen.